File: rust/examples/sdk/claim_vm_id.rs
use pwr_rs::Wallet;
use dotenvy::dotenv;
use std::env;

async fn claim() {
    dotenv().ok();
    // Setting up your wallet in the SDK
    let private_key = env::var("PRIVATE_KEY").unwrap();
    let wallet = Wallet::from_hex(&private_key).unwrap();

    // Add a unique VM ID
    let vm_id = 102032;

    // Claim the VM ID
    let tx_hash = wallet.claim_vm_id(vm_id).await;

    println!("Transaction Hash: {tx_hash}")
}

#[tokio::main]
pub async fn main() {
    claim().await;
}

File: go/sdk/claimVmId.go
package sdk

import (
	"github.com/pwrlabs/pwrgo/wallet"
	"fmt"
)

// var privateKey = "0x9D4428C6E0638331B4866B70C831F8BA51C11B031F4B55EED4087BBB8EF0151F"

func ClaimVmId() {
	// Setting up your wallet in the SDK
	wallet := wallet.FromPrivateKey(privateKey)

	// Add a unique VM ID
	vmId := 102030

	// Claim the VM ID
	tx := wallet.ClaimVMId(vmId)

	// Error handling
	if tx.Success {
		fmt.Printf("Transaction Hash: %s\n", tx.TxHash)
	} else {
		fmt.Println("Error:", tx.Error)
	}
}
